addons/binding/org.openhab.binding.upb/src/main/java/org/openhab/binding/upb/internal/UPBWriter.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.upb.internal;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.upb.internal.UPBReader.Listener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Used to write data to the UPB modem.
+ *
+ * @author Chris Van Orman
+ * @since 1.9.0
+ */
+public class UPBWriter {
+
+    /**
+     * Time in milliseconds to wait for an ACK from the modem after writing a
+     * message.
+     */
+    private static long ACK_TIMEOUT = 1000;
+
+    private static final Logger logger = LoggerFactory.getLogger(UPBWriter.class);
+
+    /**
+     * Asynchronous queue for writing data to the UPB modem.
+     */
+    // private ExecutorService executor = Executors.newSingleThreadExecutor();
+    private ExecutorService executor = new ThreadPoolExecutor(0, 1, 1000, TimeUnit.MILLISECONDS,
```

Comment:
Do we need a new ExecutorService, we already have on available from the parent class of the handler.

See code guidelines: http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior